libs/cart/src/effects/cart-resolver.effects.spec.ts

```diff
@@ -56,161 +59,285 @@ describe('DaffCartResolverEffects', () => {
-			const response = cold('#', {}, new DaffStorageServiceError());
-			driver.get.and.returnValue(response);
-			driver.create.and.returnValue(response);
-
-			const cartStorageFailureAction = new DaffCartStorageFailure('Cart Storage Failed');
-
-			actions$ = hot('--a', { a: new DaffResolveCart() });
-			const expected = cold('--b', {
-				b: cartStorageFailureAction
-			});
-
-			expect(effects.onResolveCart$).toBeObservable(expected);
-		});
-	});
+  });
+
+  describe('onResolveCart$', () => {
+    describe('handling storage errors e.g. being in SSR', () => {
+      it('should dispatch a DaffCartLoadFailure action', () => {
+        cartStorageService.getCartId.and.throwError('Storage error');
+
+        const loadCartFailureAction = new DaffCartLoadFailure(
+          'Cart loading has failed',
+        );
+
+        actions$ = hot('--a', { a: new DaffResolveCart() });
+        const expected = cold('--b', {
+          b: loadCartFailureAction
+        });
+
+        expect(effects.onResolveCart$).toBeObservable(expected);
+      });
+    });
+
+    describe('successfully resolving a cart when theres a cart id in storage', () => {
+      beforeEach(() => {
+        cartStorageService.getCartId.and.returnValue(
+          stubCart.id.toString(),
+        );
+        driver.get.and.returnValue(of(stubCart));
+      });
+
+      it('should not attempt to create a cart', () => {
+        const loadCartSuccessAction = new DaffCartLoadSuccess(stubCart);
+
+        actions$ = hot('--a', { a: new DaffResolveCart() });
+        const expected = cold('--b', {
+          b: loadCartSuccessAction,
+        });
+
+        expect(effects.onResolveCart$).toBeObservable(expected);
+        expect(driver.create).not.toHaveBeenCalled();
+      });
+
+      it('should indicate that a cart has loaded', () => {
+        const loadCartSuccessAction = new DaffCartLoadSuccess(stubCart);
+
+        actions$ = hot('--a', { a: new DaffResolveCart() });
+        const expected = cold('--b', {
+          b: loadCartSuccessAction
+        });
+
+        expect(effects.onResolveCart$).toBeObservable(expected);
+      });
+    });
+
+    describe('handling XHR errors when creating a cart', () => {
+      it('should dispatch DaffCartLoadFailure action', () => {
+        const response = cold('#', {});
+        driver.create.and.returnValue(response);
+
+        const loadCartFailureAction = new DaffCartLoadFailure(
+          'Cart loading has failed',
+        );
+
+        actions$ = hot('--a', { a: new DaffResolveCart() });
+        const expected = cold('--b', {
+          b: loadCartFailureAction
+        });
+
+        expect(effects.onResolveCart$).toBeObservable(expected);
+      });
+    });
+
+    describe('handling XHR errors when retrieving a cart', () => {
+      it('should dispatch a DaffCartLoadFailure action', () => {
+        const response = cold('#', {});
+        driver.get.and.returnValue(response);
+        driver.create.and.returnValue(response);
```

Comment:
I feel like this line shouldn't be required for the test to pass.

Reply:
If it is required, then maybe the test should be "should create a new cart", but that doesn't seem right either.

Reply:
yeah its not, removed